core/src/main/java/org/elasticsearch/index/query/MissingQueryBuilder.java

```diff
@@ -30,18 +40,26 @@
 
     public static final String NAME = "missing";
 
-    private String name;
+    public static final boolean DEFAULT_NULL_VALUE = false;
 
-    private Boolean nullValue;
+    public static final boolean DEFAULT_EXISTENCE_VALUE = true;
 
-    private Boolean existence;
+    private String fieldPattern;
+
+    private Boolean nullValue = DEFAULT_NULL_VALUE;
+
+    private Boolean existence = DEFAULT_EXISTENCE_VALUE;
 
     static final MissingQueryBuilder PROTOTYPE = new MissingQueryBuilder(null);
 
-    public MissingQueryBuilder(String name) {
-        this.name = name;
+    public MissingQueryBuilder(String fieldPattern) {
+        this.fieldPattern = fieldPattern;
```

Comment:
again, this is the reasoning: if we check for existence of a field in the parser, it means that the only way it can be null in the builder is when it comes in through java api. In that case we might want to fail fast and throw error in the constructor/setter already rather than in validate. If non validated values might come in through the parser as well then validate is the way to go. In this case it makes to do as Christoph suggested. In term query builder I think it still makes sense what we do (again, you can test it to see the differences), same for common terms query.